### ISSUE TITLE
Efficient computation of all-to-all matrix of RWR / pagerank scores

### ISSUE THREAD

**Issue by dhimmel** [sknetwork-team/scikit-network#347]
Great package! Really loving the efficiency and integration with numpy arrays / scipy.sparse matrices.

I'm interested in applying [`sknetwork.ranking.PageRank`](https://scikit-network.readthedocs.io/en/latest/reference/ranking.html#pagerank) iteratively for each node as a seed:

```python
matrix  # the input adjacency matrix
rwr_matrix = np.zeros(matrix.shape)
for i in range(nrows):
    pagerank = skn.ranking.PageRank(damping_factor=0.85)
    scores = pagerank.fit_transform(adjacency=matrix, seeds={i: 1.0})
    rwr_matrix[i] = scores  # set values in row i to contain scores
```

The output `rwr_matrix` would be a 2D matrix, with the same shape as the input adjacency matrix, but where edge weights are replaced with random walk with restart proximity scores.

I was wondering if there was a more efficient implementation that iterating through each seed node, one at a time. Is there some linear algebra that could solve this matrix all at once that sknetwork would be interested in implementing?

**Comment by tbonald** [comment]
Not that we are aware of. The transition matrix of the random walk depends on the seed(s), and the corresponding scores cannot be used for other seed(s). So a loop is needed here, as in your code.

Note that the PageRankClassifier (in the Classification algorithms) allows you to specify the seeds as labels and returns a membership matrix of shape  (#nodes, #seeds). But no miracle here, it is also based on a loop on the different labels.

Thanks!

**Comment by nathandelara** [comment]
If you have multiple cores available, you might take a look at the joblib package to speed up your loop.
https://joblib.readthedocs.io/en/latest/

Besides, be careful as your 2D output is a (n_row x n_row) **dense** matrix which my cause memory overflow for big graphs.

Thanks for your interest !
